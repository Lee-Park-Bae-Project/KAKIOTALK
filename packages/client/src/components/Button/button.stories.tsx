import React from 'react';
import { withKnobs } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import Button from './index';


export default {
  title: 'Component/Button',
  component: Button,
  decorators: [withKnobs],
  parameters: {
    componentSubtitle: '기본적인 버튼',
  },
};

export const BasicButton: React.FC = () => (
  <Button
    text='primary'
    theme='primary'
  />
);

export const SecondaryButton: React.FC = () => (
  <Button
    text='secondary'
    theme='secondary'
  />
);

export const TertiaryButton = () => (
  <Button
    text='tertiary'
    theme='tertiary'
  />
);

export const NotAllowedButton = () => (
  <Button
    text='전송'
    isAllowd={false}
  />
);

export const ButtonWithLongText: React.FC = () => (
  <Button
    text="Miusov, as a man man of breeding and deilcacy, could not but feel some inwrd qualms, when he reached the Father Superior's with Ivan: he felt ashamed"
    />
);

export const ButtonWithClickEvent: React.FC = () => (
    <Button
      text='click me'
      onClick={action('onClick')}
    />

);
