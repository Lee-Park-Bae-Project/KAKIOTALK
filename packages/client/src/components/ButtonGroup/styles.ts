import styled from 'styled-components'
import Flex from 'atoms/Flex'

interface CotnainerProps {
  direction: 'row' | 'col';
  right?: boolean;
}
const Container = styled(Flex)<CotnainerProps>`
  flex-direction: ${(props) => (props.direction === 'row' ? 'row' : 'column')};
  justify-content: ${(props) => (props.right ? 'flex-end' : 'flex-start')};
  margin:1rem;
`

export default Container
