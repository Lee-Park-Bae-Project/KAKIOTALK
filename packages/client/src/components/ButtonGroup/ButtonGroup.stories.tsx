import React from 'react';
import Button from 'components/Button';
import ButtonGroup from './index';

export default {
  title: 'Component/ButtonGroup',
  component: ButtonGroup,
};


export const DefaultButtonGroup = () => (
    <ButtonGroup>
      <Button text='hi1'/>
      <Button text='hi2'/>
    </ButtonGroup>
);

export const RightButtonGroup = () => (
  <ButtonGroup right>
    <Button text='hi1'/>
    <Button text='hi2'/>
  </ButtonGroup>
);

export const ColumnButtonGroup = () => (
  <ButtonGroup direction='col'>
    <Button text='hi1'/>
    <Button text='hi2'/>
  </ButtonGroup>
);
