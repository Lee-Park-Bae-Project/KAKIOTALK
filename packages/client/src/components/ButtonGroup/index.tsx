import React, { FC } from 'react';
import * as S from 'components/ButtonGroup/styles';

interface Props{
  /** 정렬 방향 */
  direction?: 'row' | 'col';
  /** 버튼들을 오른쪽에 정렬할 때 이용 */
  right?: boolean;
  /** 보여줄 버튼들 */
  children: React.ReactNode;
}

/**
 * 여러 버튼을 묶어서 보여줍니다.
 * children 으로 전달받은 버튼들을 나열합니다.
 */
const ButtonGroup: FC<Props> = ({
  direction = 'row',
  right = false,
  children,
}) => (
    <S.default
      direction={direction}
      right={right}
    >
      {children}
    </S.default>
);

export default ButtonGroup;
