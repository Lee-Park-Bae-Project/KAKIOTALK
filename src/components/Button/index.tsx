import React, { FC } from 'react';
import { ThemeProvider } from 'styled-components';
import * as S from 'components/Button/styles';

interface ButtonProp {
  /** 버튼안에 들어갈 텍스트 */
  text: string;
  /** theme */
  theme?: 'primary' | 'secondary' | 'tertiary';
  /** 사용가능한 버튼인지 */
  isAllowd?: boolean;
  /** 클릭 핸들러 */
  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

/**
 *
 *  - `text` 가 너무 길 경우 ... 로 표시되어 나타나지 않습니다.
 * - `onClick`에 props로 핸들러를 전달할 수 있습니다.
 */
const Button: FC<ButtonProp> = ({
  text,
  theme = 'primary',
  isAllowd = true,
  onClick,
}) => (
  <ThemeProvider theme={S.themeMap[theme]}>
    <S.Button
      type='button'
      isAllowed={isAllowd}
      onClick={onClick}
      >
      {text}
    </S.Button>
  </ThemeProvider>

);

export default Button;
