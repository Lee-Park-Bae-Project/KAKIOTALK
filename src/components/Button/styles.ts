import styled from 'styled-components';
import { darken } from 'polished';
import { color, fontSize, weight } from 'styles/global';

export const themeMap = {
  primary: {
    background: color.YELLO,
  },
  secondary: {
    background: color.RED,
  },
  tertiary: {
    background: color.GRAY,
  },
};
interface ButtonProp{
  isAllowed: boolean;
}

export const Button = styled.button<ButtonProp>`
  width: 5rem;
  height: 2rem;
  color: ${color.BLACK};
  font-size: ${fontSize.SMALL};
  font-weight: ${weight.NORMAL};

  margin: 0.5rem;

  background-color: ${(props) => props.theme.background};

  outline: none;
  border-radius: 1rem;
  border: none;
  cursor: ${(props) => (props.isAllowed ? 'pointer' : 'not-allowed')};

  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;


  &:hover{
    background-color: ${(props) => darken(0.1, props.theme.background)};
  }
`;
