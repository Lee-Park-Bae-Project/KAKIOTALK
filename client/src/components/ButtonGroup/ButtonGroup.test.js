/* eslint-disable no-undef */

import React from 'react';
import { mount } from 'enzyme';
import Button from 'components/Button';
import ButtonGroup from 'components/ButtonGroup';

describe('<ButtonnGroup />', () => {
  it('matches snapshot', () => {
    const wrapper = mount(<ButtonGroup>
      <Button text='confirm'/>
      <Button text='cancel'/>
    </ButtonGroup>);
    expect(wrapper).toMatchSnapshot();
  });
  it('Button 2개 렌더링 되는지 확인', () => {
    const wrapper = mount((
      <ButtonGroup>
        <Button text='confirm'/>
        <Button text='cancel'/>
      </ButtonGroup>
    ));

    expect(wrapper.find(Button)).toHaveLength(2);
  });

  it('Button 텍스트 확인', () => {
    const wrapper = mount((
      <ButtonGroup>
        <Button text='confirm' id='confirm-button'/>
        <Button text='cancel' id='cancel-button'/>
      </ButtonGroup>
    ));

    const confirmButton = wrapper.find('#confirm-button');
    const cancelButton = wrapper.find('#cancel-button');

    expect(confirmButton.text()).toEqual('confirm');
    expect(cancelButton.text()).toEqual('cancel');
  });
});
