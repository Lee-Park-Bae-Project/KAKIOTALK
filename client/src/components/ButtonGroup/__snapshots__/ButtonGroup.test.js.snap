// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonnGroup /> matches snapshot 1`] = `
<ButtonGroup>
  <Styled(styled.div)
    direction="row"
    right={false}
  >
    <StyledComponent
      direction="row"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "gkFfxL",
            "rules": Array [
              "
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
",
              "
  flex-direction: ",
              [Function],
              ";
  justify-content: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "Styled(styled.div)",
          "foldedComponentIds": Array [
            "sc-bwzfXH",
          ],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      right={false}
    >
      <div
        className="sc-bwzfXH sc-htpNat gkFfxL"
        direction="row"
      >
        <Button
          text="confirm"
        >
          <ThemeProvider
            theme={
              Object {
                "background": "#F7CF46",
              }
            }
          >
            <styled.button
              isAllowed={true}
              type="button"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": false,
                      "lastClassName": "eknSne",
                      "rules": Array [
                        "
  width: 5rem;
  height: 2rem;
  color: ",
                        "#4B4B4B",
                        ";
  font-size: ",
                        "0.8rem",
                        ";
  font-weight: ",
                        "600",
                        ";

  margin: 0.5rem;

  background-color: ",
                        [Function],
                        ";

  outline: none;
  border-radius: 1rem;
  border: none;
  cursor: ",
                        [Function],
                        ";

  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;


  &:hover{
    background-color: ",
                        [Function],
                        ";
  }
",
                      ],
                    },
                    "displayName": "styled.button",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "button",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                isAllowed={true}
                type="button"
              >
                <button
                  className="sc-bdVaJa eknSne"
                  type="button"
                >
                  confirm
                </button>
              </StyledComponent>
            </styled.button>
          </ThemeProvider>
        </Button>
        <Button
          text="cancel"
        >
          <ThemeProvider
            theme={
              Object {
                "background": "#F7CF46",
              }
            }
          >
            <styled.button
              isAllowed={true}
              type="button"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": false,
                      "lastClassName": "eknSne",
                      "rules": Array [
                        "
  width: 5rem;
  height: 2rem;
  color: ",
                        "#4B4B4B",
                        ";
  font-size: ",
                        "0.8rem",
                        ";
  font-weight: ",
                        "600",
                        ";

  margin: 0.5rem;

  background-color: ",
                        [Function],
                        ";

  outline: none;
  border-radius: 1rem;
  border: none;
  cursor: ",
                        [Function],
                        ";

  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;


  &:hover{
    background-color: ",
                        [Function],
                        ";
  }
",
                      ],
                    },
                    "displayName": "styled.button",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "button",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                isAllowed={true}
                type="button"
              >
                <button
                  className="sc-bdVaJa eknSne"
                  type="button"
                >
                  cancel
                </button>
              </StyledComponent>
            </styled.button>
          </ThemeProvider>
        </Button>
      </div>
    </StyledComponent>
  </Styled(styled.div)>
</ButtonGroup>
`;
